---
title: "age_binned_DE"
author: "Daniel Kiss"
date: "6/6/2023"
output: html_document
---

Load Packages etc.
```{r}
library(dplyr)
library(tidyverse)
library(edgeR)
library(markerGeneProfile) 
library(matrixStats)
library(cowplot)
library(broom)
library(knitr)
library(ggpubr)
library(biomaRt)
library(ggrepel)
library(patchwork)
library(ggsignif)
library(modelr)
library(ggbeeswarm)
library(lemon)
library(EnhancedVolcano)
theme_set(theme_classic2())
#Colour palette
cbPalette <- c("#56B4E9", "#009E73","#E69F00", "#0072B2", "#D55E00", "#CC79A7","#000000","#F0E442")
```

Load Data
```{r}
CMC_bulk = read.csv("/external/rprshnas01/kcni/dkiss/cell_prop_psychiatry/data/CMC_count_matrix.csv") 
rownames(CMC_bulk) = CMC_bulk$X
CMC_bulk[,1] = NULL

CMC_metadata = read.delim("/external/rprshnas01/netdata_kcni/stlab/PsychENCODE/CMC/Metadata/SYNAPSE_METADATA_MANIFEST.tsv") %>% subset(dataType == "geneExpression")
psychencode_metadata = read.csv(("/external/rprshnas01/netdata_kcni/stlab/PsychENCODE/Metadata/CapstoneCollection_Metadata_Clinical.csv"))
CMC_metadata = CMC_metadata %>% inner_join(psychencode_metadata)

```

Subset MSSM Data
```{r}
MSSM_metadata = CMC_metadata %>% subset(grepl(CMC_metadata$individualID, pattern = "MSSM") == TRUE & primaryDiagnosis %in% c("Schizophrenia", "control") & ageDeath <90)
#Add group name column
MSSM_metadata = MSSM_metadata %>%
  mutate(group = case_when(
    primaryDiagnosis == "Schizophrenia" & ageDeath >= 70 ~ "SCZ_Over70",
    primaryDiagnosis == "Schizophrenia" & ageDeath < 70 ~ "SCZ_Under70",
    primaryDiagnosis == "control" & ageDeath >= 70 ~ "CON_Over70",
    primaryDiagnosis == "control" & ageDeath < 70 ~ "CON_Under70",
    TRUE ~ NA_character_
  )) %>%
  mutate(newID = paste(individualID, group, sep = "_"))
#Subset CMC_bulk and add new names
MSSM_bulk = CMC_bulk[colnames(CMC_bulk) %in% MSSM_metadata$individualID]
colnames(MSSM_bulk) = MSSM_metadata$newID
```

Create DGEList and perform MDS
```{r}
#Create DGEList object and add normalization factors
d0 = DGEList(MSSM_bulk) %>% calcNormFactors()
#Filter low-expressed genes
cutoff = 4.5
drop = which(apply(cpm(d0), 1, max) < cutoff)
d = d0[-drop,] 
dim(d) # number of genes left
#Get names for groups

group_names = c()
for (col_name in colnames(MSSM_bulk)) {
  group_names = c(group_names, substr(col_name, nchar(col_name) - 10, nchar(col_name)))
} 
group_names = group_names %>% as.factor
#Plot MDS to visualize group clustering 
MDS = plotMDS(d, col = as.numeric(group_names))
mds_coords = MDS$cmdscale.out
mds_data <- data.frame(x = mds_coords[, 1], y = mds_coords[, 2], group = group_names)

# Create the MDS plot using ggplot2
ggplot(mds_data, aes(x = x, y = y, color = group)) +
  geom_point(size = 4) +
  labs(x = "MDS1", y = "MDS2") +
  theme_minimal() +
  theme(legend.position = "bottom")

```

Voom transformation and calculation of variance weights
```{r}
#Specify the model to be fitted. We do this before using voom since voom uses variances of the model residuals (observed - fitted)
mm <- model.matrix(~0 + group_names)
y <- voom(d, mm, plot = T)
#A good voom plot will have very few points at X = 0
```

CON vs. SCZ: Under 70
```{r}
#lmFit fits a linear model using weighted least squares for each gene:
fit <- lmFit(y, mm)
#Comparison between times 6 and 9 for cultivar I5
contr <- makeContrasts(group_namesCON_Under70 - group_namesSCZ_Under70, levels = colnames(coef(fit)))
contr
#Estimate contrast for each gene
tmp <- contrasts.fit(fit, contr)
#Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp <- eBayes(tmp)
#What genes are most differentially expressed?
DE_under_70 <- topTable(tmp, sort.by = "P", n = Inf)
head(DE_under_70, 20)
#How many DE genes are there?
length(which(DE_under_70$adj.P.Val < 0.05))
#Write top.table to a file
DE_under_70$Gene <- rownames(DE_under_70)
DE_under_70 <- DE_under_70[,c("Gene", names(DE_under_70)[1:6])]
```

CON vs. SCZ: Over 70
```{r}
#lmFit fits a linear model using weighted least squares for each gene:
fit <- lmFit(y, mm)
#Comparison between times 6 and 9 for cultivar I5
contr <- makeContrasts(group_names_CON_Over70 - group_names_SCZ_Over70, levels = colnames(coef(fit)))
contr
#Estimate contrast for each gene
tmp <- contrasts.fit(fit, contr)
#Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp <- eBayes(tmp)
#What genes are most differentially expressed?
DE_over_70 <- topTable(tmp, sort.by = "P", n = Inf)
head(DE_over_70, 20)
#How many DE genes are there?
length(which(DE_over_70$adj.P.Val < 0.05))
#Write top.table to a file
DE_over_70$Gene <- rownames(DE_over_70)
DE_over_70 <- DE_over_70[,c("Gene", names(DE_over_70)[1:6])]
```

```{r}
tplot_bulk = merge(DE_over_70, DE_under_70, by = 0) %>% 
  ggplot(aes(x = t.x, y = t.y)) + 
  geom_point(size = 1) +
  ylab("Over 70") +
  xlab("Under 70") +
  ggtitle("Bulk t-Statistics")
tplot_bulk
```


//// An attempt at running the above pipeline w/ psedobulked scRNAseq data ////


Load Data
```{r}
pseudobulk = read.delim("/external/rprshnas01/netdata_kcni/stlab/SZBDMulticohort/pseudobulk_fine_grained.counts.tsv")
#Pivot row to col
pseudobulk = as.data.frame(t(pseudobulk))
colnames(pseudobulk) =  pseudobulk[1, ]
pseudobulk = pseudobulk[-1, ]
pseudobulk_metadata = read.delim("/external/rprshnas01/netdata_kcni/stlab/SZBDMulticohort/pseudobulk_fine_grained.metadata.tsv")
```

Subset Data
```{r}
pseudobulk_metadata = pseudobulk_metadata %>%
  subset(grepl(pseudobulk_metadata$CMC_ID, pattern = "MSSM") == TRUE) %>%
  mutate(group = case_when(
    Phenotype == "SZ" & Age >= 70 ~ "SCZ_Over70 ",
    Phenotype == "SZ" & Age < 70 ~ "SCZ_Under70",
    Phenotype == "CON" & Age >= 70 ~ "CON_Over70 ",
    Phenotype == "CON" & Age < 70 ~ "CON_Under70",
    TRUE ~ NA_character_
  )) %>%
  mutate(newID = paste(CMC_ID, group, sep = "_"))
#Subset and add new names
pseudobulk_metadata = pseudobulk_metadata %>% subset(pseudobulk_metadata$Cohort == "MtSinai")
pseudobulk = pseudobulk[colnames(pseudobulk) %in% pseudobulk_metadata$unique_donor_ID]
colnames(pseudobulk) = pseudobulk_metadata$newID
```

Create individual dataframes for each cell type (SST, PVALB, VIP)
```{r}
pseudobulk_SST = pseudobulk[which(pseudobulk[1, ] == "In-SST")][-1,] %>% lapply(as.numeric) %>% as.data.frame()
row.names(pseudobulk_SST) = row.names(pseudobulk[-1, ])
pseudobulk_VIP = pseudobulk[which(pseudobulk[1, ] == "In-VIP")][-1,] %>% lapply(as.numeric) %>% as.data.frame()
row.names(pseudobulk_VIP) = row.names(pseudobulk[-1, ])
```

SST: Create DGEList and perform MDS
```{r}
#Create DGEList object and add normalization factors
d0_SST = DGEList(pseudobulk_SST) %>% calcNormFactors()
#Filter low-expressed genes
cutoff_SST = 20
drop_SST = which(apply(cpm(d0_SST), 1, max) < cutoff_SST)
d_SST = d0_SST[-drop_SST,] 
dim(d_SST) # number of genes left
#Get names for groups

group_names_SST = c()
for (col_name in colnames(pseudobulk_SST)) {
  group_names_SST = c(group_names_SST, substr(col_name, nchar(col_name) - 10, nchar(col_name)))
} 
group_names_SST = group_names_SST %>% as.factor
#Plot MDS to visualize group clustering 
MDS_SST = plotMDS(d_SST, col = as.numeric(group_names_SST))
mds_coords_SST = MDS_SST$cmdscale.out
mds_data_SST <- data.frame(x = mds_coords_SST[, 1], y = mds_coords_SST[, 2], group = group_names_SST)

# Create the MDS plot using ggplot2
ggplot(mds_data_SST, aes(x = x, y = y, color = group)) +
  geom_point(size = 4) +
  labs(x = "MDS1", y = "MDS2") +
  theme_minimal() +
  theme(legend.position = "bottom")

```

SST: Voom transformation and calculation of variance weights
```{r}
#Create factors for model covariates
sex = pseudobulk_metadata %>% filter(Celltype == "In-SST") %>% pull(Gender)
pmi = pseudobulk_metadata %>% filter(Celltype == "In-SST") %>% pull(PMI)
age = pseudobulk_metadata %>% filter(Celltype == "In-SST") %>% pull(Age)
cells_per_donor = log10(pseudobulk_metadata %>% filter(Celltype == "In-SST") %>% pull(num_cells))

#Specify the model to be fitted. We do this before using voom since voom uses variances of the model residuals (observed - fitted)
mm_SST <- model.matrix(~0 + group_names_SST + age + pmi + sex + cells_per_donor)
y_SST <- voom(d_SST, mm_SST, plot = T)
#A good voom plot will have very few points at X = >0
```

SST: CON vs. SCZ: UNDER 70
```{r, fig.width=4, fig.height=4}
#lmFit fits a linear model using weighted least squares for each gene:
fit_SST <- lmFit(y_SST, mm_SST)
#Comparison between times 6 and 9 for cultivar I5
contr_SST <- makeContrasts(group_names_SSTCON_Under70 - group_names_SSTSCZ_Under70, levels = colnames(coef(fit_SST)))
contr_SST
#Estimate contrast for each gene
tmp_SST <- contrasts.fit(fit_SST, contr_SST)
#Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp_SST <- eBayes(tmp_SST)
#What genes are most differentially expressed?
DE_under_70_SST <- topTable(tmp_SST, sort.by = "P", n = Inf)
head(DE_under_70_SST, 20)
#How many DE genes are there?
length(which(DE_under_70_SST$adj.P.Val < 0.05))

#Volcano Plot
EnhancedVolcano(DE_under_70_SST,
                lab = rownames(DE_under_70_SST),
                x = 'logFC',
                y = 'P.Value')

```

SST: CON vs. SCZ: OVER 70
```{r, fig.width=4, fig.height=4}
#lmFit fits a linear model using weighted least squares for each gene:
fit_SST <- lmFit(y_SST, mm_SST)
#Comparison between times 6 and 9 for cultivar I5
contr_SST <- makeContrasts(group_names_SSTCON_Over70. - group_names_SSTSCZ_Over70., levels = colnames(coef(fit_SST)))
contr_SST
#Estimate contrast for each gene
tmp_SST <- contrasts.fit(fit_SST, contr_SST)
#Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp_SST <- eBayes(tmp_SST)
#What genes are most differentially expressed?
DE_over_70_SST <- topTable(tmp_SST, sort.by = "P", n = Inf)
head(DE_over_70_SST, 20)
#How many DE genes are there?
length(which(DE_over_70_SST$adj.P.Val < 0.05))

#Volcano Plot
EnhancedVolcano(DE_over_70_SST,
                lab = rownames(DE_under_70_SST),
                x = 'logFC',
                y = 'P.Value')
```

SST: Plot t-statistics from both analyses
```{r}
tplot_SST = merge(DE_over_70_SST, DE_under_70_SST, by = 0) %>% 
  ggplot(aes(x = t.x, y = t.y)) + 
  geom_point(size = 1) +
  ylab("Over 70") +
  xlab("Under 70") +
  ggtitle("SST t-Statistics")
tplot_SST

```

VIP: Create DGEList and perform MDS
```{r}
# Create DGEList object and add normalization factors
d0_VIP = DGEList(pseudobulk_VIP) %>% calcNormFactors()
# Filter low-expressed genes
cutoff_VIP = 1
drop_VIP = which(apply(cpm(d0_VIP), 1, max) < cutoff_VIP)
d_VIP = d0_VIP[-drop_VIP,] 
dim(d_VIP) # number of genes left
# Get names for groups
group_names_VIP = c()
for (col_name in colnames(pseudobulk_VIP)) {
  group_names_VIP = c(group_names_VIP, substr(col_name, nchar(col_name) - 10, nchar(col_name)))
} 
group_names_VIP = group_names_VIP %>% as.factor
# Plot MDS to visualize group clustering 
MDS_VIP = plotMDS(d_VIP, col = as.numeric(group_names_VIP))
mds_coords_VIP = MDS_VIP$cmdscale.out
mds_data_VIP <- data.frame(x = mds_coords_VIP[, 1], y = mds_coords_VIP[, 2], group = group_names_VIP)

# Create the MDS plot using ggplot2
ggplot(mds_data_VIP, aes(x = x, y = y, color = group)) +
  geom_point(size = 4) +
  labs(x = "MDS1", y = "MDS2") +
  theme_minimal() +
  theme(legend.position = "bottom")
```

VIP: Voom transformation and calculation of variance weights
```{r}
# Create factors for model covariates
sex = pseudobulk_metadata %>% filter(Celltype == "In-VIP") %>% pull(Gender)
pmi = pseudobulk_metadata %>% filter(Celltype == "In-VIP") %>% pull(PMI)
age = pseudobulk_metadata %>% filter(Celltype == "In-VIP") %>% pull(Age)
cells_per_donor = log10(pseudobulk_metadata %>% filter(Celltype == "In-VIP") %>% pull(num_cells))

# Specify the model to be fitted. We do this before using voom since voom uses variances of the model residuals (observed - fitted)
mm_VIP <- model.matrix(~0 + group_names_VIP + age + pmi + sex + cells_per_donor)
y_VIP <- voom(d_VIP, mm_VIP, plot = T)
# A good voom plot will have very few points at X = >0
```

VIP: CON vs. SCZ: UNDER 70
```{r, fig.width=4, fig.height=4}
# lmFit fits a linear model using weighted least squares for each gene:
fit_VIP <- lmFit(y_VIP, mm_VIP)
# Comparison between times 6 and 9 for cultivar I5
contr_VIP <- makeContrasts(group_names_VIPCON_Under70 - group_names_VIPSCZ_Under70, levels = colnames(coef(fit_VIP)))
contr_VIP
# Estimate contrast for each gene
tmp_VIP <- contrasts.fit(fit_VIP, contr_VIP)
# Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp_VIP <- eBayes(tmp_VIP)
# What genes are most differentially expressed?
DE_under_70_VIP <- topTable(tmp_VIP, sort.by = "P", n = Inf)
head(DE_under_70_VIP, 20)
# How many DE genes are there?
length(which(DE_under_70_VIP$adj.P.Val < 0.05))

# Volcano Plot
EnhancedVolcano(DE_under_70_VIP,
                lab = rownames(DE_under_70_VIP),
                x = 'logFC',
                y = 'P.Value')
```

VIP: CON vs. SCZ: OVER 70
```{r, fig.width=4, fig.height=4}
# lmFit fits a linear model using weighted least squares for each gene:
fit_VIP <- lmFit(y_VIP, mm_VIP)
# Comparison between times 6 and 9 for cultivar I5
contr_VIP <- makeContrasts(group_names_VIPCON_Over70. - group_names_VIPSCZ_Over70., levels = colnames(coef(fit_VIP)))
contr_VIP
# Estimate contrast for each gene
tmp_VIP <- contrasts.fit(fit_VIP, contr_VIP)
# Empirical Bayes smoothing of standard errors (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error) (see https://www.degruyter.com/doi/10.2202/1544-6115.1027)
tmp_VIP <- eBayes(tmp_VIP)
# What genes are most differentially expressed?
DE_over_70_VIP <- topTable(tmp_VIP, sort.by = "P", n = Inf)
head(DE_over_70_VIP, 20)
# How many DE genes are there?
length(which(DE_over_70_VIP$adj.P.Val < 0.05))

# Volcano Plot
EnhancedVolcano(DE_over_70_VIP,
                lab = rownames(DE_under_70_VIP),
                x = 'logFC',
                y = 'P.Value')

```

VIP: Plot t-statistics from both analyses
```{r}
tplot_VIP = merge(DE_over_70_VIP, DE_under_70_VIP, by = 0) %>% 
  ggplot(aes(x = t.x, y = t.y)) + 
  geom_smooth() +
  geom_point(size = 1, alpha = 0.05) +
  ylab("Over 70") +
  xlab("Under 70") +
  ggtitle("VIP t-Statistics")
tplot_VIP
```

Next Steps:
Are sexMale DEs correlated between age groups?
Compare McLean and MSSM to each other within age groups

